lib/matplotlib/colors.py

```diff
@@ -225,9 +225,14 @@ def _to_rgba_no_colorcycle(c, alpha=None):
             return tuple(color)
         # string gray.
         try:
-            return (float(c),) * 3 + (alpha if alpha is not None else 1.,)
+            c = float(c)
         except ValueError:
             pass
+        else:
+            if not (0 <= c <= 1):
+                raise ValueError(
+                    "String grayscale values must be within 0-1 range")
```

Comment:
All other ValueErrors report back the `orig_c`. Would it make sense to do so here as well?
Possibly:

    "Invalid string grayscale value {!r}. Values must be within 0-1 range.".format(orig_c)